### PR TITLE
React port

### PR DESCRIPTION
**Pull request by paulpugh000**
rewrote the vanilla JavaScript into React, with use of React hooks and the fetch API instead of the jQuery AJAX method

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,12 +8,16 @@
...
   },
-  "devDependencies": {
-    "vite": "^4.2.0"
-  },
   "dependencies": {
-    "jquery": "^3.6.4",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
     "ress": "^5.0.2",
     "toastify-js": "^1.12.0"
```
Comment (NRProjects) [inline package.json:15]: (⁉) Are these dependencies really needed?

File: package.json [hunk 2]
```diff
@@ -8,12 +8,16 @@
...
+  },
+  "devDependencies": {
+    "@types/react": "^18.0.28",
+    "@types/react-dom": "^18.0.11",
+    "@vitejs/plugin-react-swc": "^3.0.0",
+    "vite": "^4.2.0"
```
Comment (NRProjects) [inline package.json:21]: (🔧) No references to drugs in this professional learning environment

File: src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+function getWeatherIcon(weatherCondition) {
+  const weatherIconMap = {
+      "Clear": "animated/day.svg",
+      "Clouds": "animated/cloudy.svg",
+      "Thunderstorm": "animated/thunder.svg",
+      "Drizzle": "animated/rainy-1.svg",
+      "Rain": "animated/rainy-7.svg",
+      "Snow": "animated/snowy-6.svg"
+  };
```
Comment (NRProjects) [inline src/App.jsx:18]: (✔) Really great map. 👍🏿👍🏿👍🏿

File: package.json (hunk 1 above)
Comment (paulpugh000) [reply package.json:15]: react and react-dom are dependencies for the React framework, of which this rewrite was built upon.
ress is a css reset to enable more precise control of CSS styling, and toastify-js adds a simple toast component to add notifications to users.

File: src/App.jsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  const [search, setSearch] = useState("ENTER LOCATION");
+  const [weather, setWeather] = useState({
+    temperature:"0",
+    graphic: "",
+    graphicAlt: "",
+    feelsLikeTemp:"",
+    humidity:"0",
+    wind:"0",
+    location:"",
+    weatherDescription:"",
```
Comment (NRProjects) [inline src/App.jsx:32]: (👖) Linting error. Please add spaces after colons or at least be consistent

File: src/App.jsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+      "Rain": "animated/rainy-7.svg",
+      "Snow": "animated/snowy-6.svg"
+  };
+  return weatherIconMap[weatherCondition] || "animated/weather.svg";
+}
+
+function App() {
+  const [search, setSearch] = useState("ENTER LOCATION");
+  const [weather, setWeather] = useState({
+    temperature:"0",
```
Comment (NRProjects) [inline src/App.jsx:25]: (👖) Linting error. Please add spaces after colons or at least be consistent

File: src/App.jsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+          <img src={windGust} alt="Wind Gust"/>
+          <p id="wind">{weather.wind} mp/h</p>
+          <p className="wind-label">Wind Speed</p>
+        </div>
+      </div>
+    </div>
+  )
+}
+
+export default App
```
Comment (NRProjects) [inline src/App.jsx:135]: (🐛)

```javascript
import React, { useState, useCallback, memo } from 'react';

// ... other imports ...

const WeatherBox = memo(function WeatherBox({ weather }) {
  return (
    <div className="weather-box">
      <img
        id="weather-graphic"
        src={weather.graphic}
        alt={weather.graphicAlt}
      />
      <p id="weather-description" className="weight-thin gray">
        {weather.weatherDescription}
      </p>
    </div>
  );
});

function App() {
  // ... useState code ...

  const handleSubmit = useCallback(
    (e) => {
      // ... same as before ...
    },
    [search, weather]
  );

  // ... return JSX code ...
  return (
    <div className="center">
      <div className="container">
        {/* ... other components ... */}
        <WeatherBox weather={weather} />
        {/* ... other components ... */}
      </div>
    </div>
  );
}

export default App;
```

ChatGPT 4 said to use this so you're probably bloating this and doing it wrong.

**Review by NRProjects — CHANGES_REQUESTED** [review]
Please address the biggest issues I've listed

File: src/App.jsx (hunk 6 above)
Comment (paulpugh000) [reply src/App.jsx:135]: This doesn't include any of the actual working parts of the program, just a bunch of scaffolding with comments.

**Pull request merged**